Tests/ApplicativeRouterHttpPipelineSupportTests/ApplicativeRouterHttpPipelineSupportTests.swift

```diff
@@ -12,10 +12,17 @@ class ApplicativeRouterHttpPipelineSupportTests: XCTestCase {
       Route.iso.home <¢> get <% end
         <|> Route.iso.episode <¢> get %> lit("episode") %> .string <% end
 
-    let middleware: Middleware<StatusLineOpen, ResponseEnded, Prelude.Unit, Data> =
+    let middleware: Middleware<StatusLineOpen, ResponseEnded, Never, Never, Prelude.Unit, Data> =
       route(router: router)
         <| writeStatus(.ok)
-        >-> { $0 |> respond(text: "Recognized route: \($0.data)") }
+
+        >>> map(map { Data("Recognized route: \($0)".utf8) })
+        // FIXME: for snapshot to render body
+        >-> writeHeader(.contentType(.plain))
+        >-> { $0 |> writeHeader(.contentLength($0.data.right!.count)) }
```

Comment:
These two lines are here because `end` doesn't set those headers. We should discuss the clumsiness here in general.